### PR TITLE
Participant Records Yesterday's Emotions

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Create provider (& spec) to allow participants to record past emotions
* Add view to provide the adding of 23 specific emotions
* Add feature spec to verify participants can save emotions
* Add js specs to `createEmotionalRating`

[#114004809]

### CODE REVIEW

File: app/assets/javascripts/think_feel_do_engine/feel/create_emotional_ratings.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            emotion_id: $tr.data('emotionId'),
+            is_positive: $tr.data('isPositive'),
+            participant_id: options.participant_id,
+            rating: rating
+          }
+        },
+        async: true,
+        dataType: 'script',
+        success: function(response) {
+          redirect(options);
```
Comment (eschlange) [inline app/assets/javascripts/think_feel_do_engine/feel/create_emotional_ratings.js]: If redirect only occurs on a success, is there any indication of failure when there it is not successful?

File: app/models/content_providers/emotions/new_yesterday_provider.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
+module ContentProviders
+  module Emotions
+    # Allows participants to record all of yesterday's emotions
+    class NewYesterdayProvider < BitCore::ContentProvider
+      Description = Struct.new(:name, :is_positive)
```
Comment (eschlange) [inline app/models/content_providers/emotions/new_yesterday_provider.rb]: Nice usage of Struct!

File: app/assets/javascripts/think_feel_do_engine/feel/create_emotional_ratings.js (hunk 1 above)
Comment (michaelwehrley) [inline app/assets/javascripts/think_feel_do_engine/feel/create_emotional_ratings.js]: good catch.

**Comment by eschlange** [comment]
Good test coverage, I say :shipit: after addressing the one issue that I pointed out.

File: spec/dummy/spec/javascripts/think_feel_do_engine/feel/create_emotional_rating_spec.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      beforeEach(function() {
+        options.rows_of_emotions = $('' +
+          '<tr>' +
+            '<td>' +
+              '<input type="radio" value="0" checked>' +
+            '</td>' +
+          '</tr>');
+      });
+
+      describe('when selected choice exists', function() {
```
Comment (michaelwehrley) [inline spec/dummy/spec/javascripts/think_feel_do_engine/feel/create_emotional_rating_spec.js]: @eschlange could you just give a look here at the `success` and `failure` responses.  Thanks!

Comment (eschlange) [inline spec/dummy/spec/javascripts/think_feel_do_engine/feel/create_emotional_rating_spec.js]: Looks good! I like the mocked Ajax responses.

**Pull request merged**